### PR TITLE
Feature/key models

### PR DESCRIPTION
**Pull request by joshbax189**
Fixes a bug reported by Zak & Vijay: UTF-8 strings must be encrypted with the correct encoding, but keys should always be encrypted with no encoding (or at least the same encoding on both encryption and decryption). On further inspection, there are 3 types of key which are passed around the SDK (4 if you consider serialized/raw keys too) and all described as "string". 

I applied the same pattern as for decrypted-item PR: add SymmetricKey and DecryptedKeypair to wrap keys in the SDK, and all cryppo.X methods become methods on the respective key classes. These are also typed, so you generally do the right thing. Deserialization is also checked, so you don't deserialize a raw key, or pass a public key as a private key etc.

### CODE REVIEW

File: packages/sdk/src/models/decrypted-keypair.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * Encryption and decryption methods are defined on the classes {@link RSAPublickey} and {@link RSAPrivatekey} which may
+ * be used if only one key is available.
+ */
+export default class DecryptedKeypair {
+  publicKey: RSAPublicKey;
+  privateKey: RSAPrivateKey;
+
+  /** A new random keypair with default length. */
+  static async new(keyBits = 4096): Promise<DecryptedKeypair> {
```
Comment (zbarbuto) [inline packages/sdk/src/models/decrypted-keypair.ts:18]: Nit: I probably prefer `generate` or `generateNew` over `new` - `new` sort of implies you can pass in private/public keys to create a new instance of the key pair.

File: packages/sdk/src/models/decrypted-keypair.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    const { privateKey, publicKey } = await cryppo.generateRSAKeyPair(keyBits);
+    return new DecryptedKeypair(publicKey, privateKey);
+  }
+
+  /**
+   * Decrypts a keypair from the Meeco Keystore.
+   * @param masterKey Usually the user's key encryption key.
```
Comment (zbarbuto) [inline packages/sdk/src/models/decrypted-keypair.ts:25]: Should probably be `keyEncryptionKey` then. `masterKey` generally refers to the passphrase derived key (albeit the name is now deprecated)

**Comment by zbarbuto** [comment]
I have a general worry that some concerns that are mostly cryppo-specific are leaking into the js-sdk too much.

For example, the classes `DecryptedKeypair` `RSAPrivateKey` `RSAPublicKey` and `SymmetricKey` are all (mostly) pretty much just wrappers around built-in functionality in Cryppo to make it a bit more ergonomic to work with. If it's the case that we need all these helpers to deal with Cryppo I think it's possibly a good indication that these classes should just come out-of-the-box in Cryppo. The functional style methods can remain but having the class wrappers there as an optional addition seems like a good idea (the other Cryppo ports provide something similar).

The main exception is the key pair `fromAPI` method which I think could just be updated to use a plain type that conforms to the API type (we want to avoid adding API sdk's as dependencies of cryppo).

The bright side is that this would be a pretty simple refactor as you could just pull those ~4 files into Cryppo and find-and-replace the imports here.

**Comment by joshbax189** [comment]
> I have a general worry that some concerns that are mostly cryppo-specific are leaking into the js-sdk too much.

Fair point.

> For example, the classes `DecryptedKeypair` `RSAPrivateKey` `RSAPublicKey` and `SymmetricKey` are all (mostly) pretty much just wrappers around built-in functionality in Cryppo to make it a bit more ergonomic to work with. If it's the case that we need all these helpers to deal with Cryppo I think it's possibly a good indication that these classes should just come out-of-the-box in Cryppo. The functional style methods can remain but having the class wrappers there as an optional addition seems like a good idea (the other Cryppo ports provide something similar).

The way I understand it is that cryppo supports n encryption options, and m serialization options, but the js-sdk is only interested in the few that the API actually requires. Specifically it should fix the cryppo defaults it depends on within the js-sdk codebase, to avoid stealth dependencies. 

That would mean classes ported over to cryppo would take key/encryption/serialization params as part of their constructors right? But then the js-sdk would have some class `SDKKey` fixing those params again.

It would help to have those classes provided out of the box though; and BW is an instructive example, with crypto logic sprinkled everywhere and loads of strings named 'key' flying around... it's not the best.

File: packages/sdk/src/models/decrypted-keypair.ts (hunk 1 above)
Comment (joshbax189) [reply packages/sdk/src/models/decrypted-keypair.ts:18]: Eugh that's a whole 8 more characters! Will change it though. Would it make sense to also have other constructor args here though: `DecryptedKeypair.generate(keyBits, externalIds)`

Comment (zbarbuto) [reply packages/sdk/src/models/decrypted-keypair.ts:18]: Only if it's useful

> Eugh that's a whole 8 more characters
You've done all the hard work of wiring it up now so it's a simple F2 to change it to the more verbose version :D

**Review by zbarbuto — APPROVED** [review]

**Pull request merged**